packages/core/src/tracing/cls.js

```diff
@@ -10,39 +10,72 @@ const tracingUtil = require('./tracingUtil');
 const { ENTRY, EXIT, INTERMEDIATE } = require('./constants');
 const hooked = require('./clsHooked');
 const tracingMetrics = require('./metrics');
+
+/**
+ * @typedef {Object} GetFromProperties
+ * @property {string} e
+ * @property {string} h
+ */
+
+/**
+ * @typedef {Function} GetFromFunction
+ * @returns {GetFromProperties}
+ */
+
+/**
+ * @description This object is actually here: /nodejs-sensor/packages/collector/src/pidStore/index.js
+ * @typedef {Object} PIDStore
+ * @property {GetFromFunction} getFrom
+ */
+
+/** @type {import('../logger').GenericLogger} */
 let logger;
 logger = require('../logger').getLogger('tracing/cls', newLogger => {
   logger = newLogger;
 });
 
-const currentEntrySpanKey = (exports.currentEntrySpanKey = 'com.instana.entry');
-const currentSpanKey = (exports.currentSpanKey = 'com.instana.span');
-const reducedSpanKey = (exports.reducedSpanKey = 'com.instana.reduced');
-const tracingLevelKey = (exports.tracingLevelKey = 'com.instana.tl');
-const w3cTraceContextKey = (exports.w3cTraceContextKey = 'com.instana.w3ctc');
+const currentEntrySpanKey = 'com.instana.entry';
+const currentSpanKey = 'com.instana.span';
+const reducedSpanKey = 'com.instana.reduced';
+const tracingLevelKey = 'com.instana.tl';
+const w3cTraceContextKey = 'com.instana.w3ctc';
 
 // eslint-disable-next-line no-undef-init
-let serviceName = undefined;
+/** @type {String} */
+let serviceName;
+/** @type {PIDStore} */
 let processIdentityProvider = null;
 
-/*
- * Access the Instana namespace in continuation local storage.
- *
- * Usage:
- *   cls.ns.get(key);
- *   cls.ns.set(key);
- *   cls.ns.run(function() {});
+/**
+ * This type is to be used across the code base, as we don't have a public explicit type for a span.
+ * Also, it is often that we simply create a literal object and gradually throw few properies on it and
+ * handle them as spans. This type has all span properties, but they are all optional, so we can safely
+ * type these literal objects.
+ * TODO: move InstanaSpan and InstanaPseudoSpan to their own file and make them publicly accessible?
```

Comment:
@basti1302 I would like to discuss this with you. Not sure if there is a reason for the InstanaSpan and InstanaPseudoSpan to be private within the module

Reply:
Ok. I won't touch this right now, for the sake of not making this PR even fatter. But will do in the next iteration.